src/braket/circuits/circuit.py

```diff
@@ -438,6 +440,136 @@ def add_circuit(
 
         return self
 
+    def add_noise(
+        self,
+        noise: Noise,
+        target_gates: Union[str, Iterable[str]] = None,
+        target_qubits: QubitSetInput = None,
+        target_times: Union[int, Iterable[int]] = None,
+        insert_strategy: str = "inclusive",
+    ) -> Circuit:
+        """
+        Add the provided `noise` to gates, qubits and time specified by `target_gates`,
+         `target_qubits` and `target_times`. When `target_gates`=None, `noise` is added
+        to the specified qubits and time. When `target_qubits`=None, `noise` is added to
+        the specified gates and time at all qubits. When `target_times`=None, `noise` is
+        added to the specified gates and qubits at all time. If all `target_gates`,
+        `target_qubits` and `target_times` are None, `noise` is added to every qubit at
+        every time in a circuit.
+
+        When `target_gates` is not None and `noise.qubit_count`>1, `noise.qubit_count`
+        must be the same as `qubit_count` of gates specified by `target_gates`. When
+        `noise.qubit_count`==1, ie. `noise` is a single-qubit noise, `noise` is added
+        to all qubits in `target_qubits` that interact with the target gates.
+
+        Args:
+            noise (Noise): Noise to be added to the circuit. When `noise.qubit_count`>1,
+                `noise.qubit_count` must be the same as `qubit_count` of gates specified by
+                `target_gates`.
+            targer_gates (Union[str, Iterable[str]): Name or List of name of gates which
```

Comment:
+1

Reply:
Just to confirm, do you mean the class themselves (without initialize, like Noise.BitFlip) or the class object (initialized, like Noise.BitFlip())?

Reply:
I was thinking the class themselves (without initializing, like Gate.X), but hadn't given passing in the objects any thought. Do you see a strong advantage in passing in the objects instead ?

Reply:
The gates are also passed in as objects for instructions. The reason they are passed in as objects, is to allow passing in of variables to the object initiation, like the angle to AngledGate.

Reply:
I agree, I also need the qubit_count of a gate for validation before actually adding noise. The concern I have is that, when users pass in an object as `target_gates`, they need to specify parameters (e.g. angle) that is not used anywhere because only the type of the gate is compared.